dev/dev-env-setup.sh

```diff
@@ -0,0 +1,233 @@
+pyenv exec pip install $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) --upgrade pip
+pyenv exec pip install $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) virtualenv
+
+VENV_DIR="$directory/bin/activate"
+
+# Check if the virtualenv already exists at the specified path
+if [[ -d "$directory"  ]]; then
+  if [ -z "$MLFLOW_DEV_ENV_REPLACE_ENV" ]; then
+    read -p "A virtual environment is already located at $directory. Do you wish to replace it? $(tput bold; tput setaf 2)(y/n) $(tput sgr0)" -n 1 -r
+    echo
+  fi
+  if [[ $REPLY =~ ^[Yy]$ || $MLFLOW_DEV_ENV_REPLACE_ENV == 1 ]]; then
+    deactivate
+    rm -rf "$directory"
+    echo "Virtual environment removed from '$directory'. Installing new instance."
+    pyenv exec virtualenv "$directory"
+  fi
+else
+  # Create a virtual environment with the specified Python version
+  pyenv exec virtualenv "$directory"
+fi
+
+# Activate the virtual environment
+# shellcheck disable=SC1090
+source "$VENV_DIR"
+
+echo "$(tput setaf 2)Current Python version: $(tput bold)$(python --version)$(tput sgr0)"
+echo "$(tput setaf 3)Activated environment is located: $(tput bold) $directory/bin/activate$(tput sgr0)"
+
+echo "Installing pip dependencies for development environment."
+
+if [[ $full_install == 1 ]]; then
+  # Install required dependencies for Prophet
+  tmp_dir=$(mktemp -d)
+  pip download $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) --no-deps --dest "$tmp_dir" --no-cache-dir prophet
+  tar -zxvf "$tmp_dir"/*.tar.gz -C "$tmp_dir"
+  pip install $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) -r "$(find "$tmp_dir" -name requirements.txt)"
+  rm -rf "$tmp_dir"
+  # Install dev requirements and test plugin
+  pip install $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) -r "$MLFLOW_HOME/requirements/dev-requirements.txt"
+  # Install current checked out version of MLflow (local)
+  pip install $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) -e .[extras]
+  # Install test plugin
+  pip install $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) -e "$MLFLOW_HOME/tests/resources//mlflow-test-plugin"
+  echo "Finished installing pip dependencies."
+else
+  pip install $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) -r "$MLFLOW_HOME/requirements/skinny-requirements.txt"
+  pip install $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) -r "$MLFLOW_HOME/requirements/small-requirements.txt"
+  pip install $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) -r "$MLFLOW_HOME/requirements/lint-requirements.txt"
+  pip install $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) -r "$MLFLOW_HOME/requirements/large-requirements.txt"
+  pip install $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' ) -r "$MLFLOW_HOME/requirements/doc-requirements.txt"
+fi
+
+command -v docker >/dev/null 2>&1 || echo "$(tput bold; tput setaf 1)A docker installation cannot be found. Please ensure that docker is installed to run all tests locally.$(tput sgr0)"
```

Comment:
Can we also support installing docker in the script ?

Reply:
We also have some tests rely on kubernetes. Shall we also support installing and set-up kubernetes in the script ?

Reply:
It is fine to start with support docker/kubernetes installation only in mac.

Reply:
I don't think we need to setup docket or kubenetes because most of contributors don't need to touch them.

Reply:
The more things we try to setup, the harder it is for this script to succeed.

Reply:
We can provide an option to install docker / kubernets and by default the option is off.
We can do it in follow-up PR.

Reply:
Sounds good. Do you want to work on that?

Reply:
I would help to work on that when I have free time.